File: server/datasets/python_source-code/Multiple-G1/renamed_44_bst.py
class Node_8:
    def __init__(self, val_8):
        self.val_8 = val_8
        self.lhs_8 = None
        self.rhs_8 = None

class Tree_8:
    def __init__(self):
        self.rt_8 = None

    def ins_8(self, val_8):
        if self.rt_8 is None:
            self.rt_8 = Node_8(val_8)
        else:
            self._ins_8_rec(self.rt_8, val_8)

    def _ins_8_rec(self, nd_8, val_8):
        if val_8 < nd_8.val_8:
            if nd_8.lhs_8 is None:
                nd_8.lhs_8 = Node_8(val_8)
            else:
                self._ins_8_rec(nd_8.lhs_8, val_8)
        elif val_8 > nd_8.val_8:
            if nd_8.rhs_8 is None:
                nd_8.rhs_8 = Node_8(val_8)
            else:
                self._ins_8_rec(nd_8.rhs_8, val_8)

    def srch_8(self, val_8):
        return self._srch_8_rec(self.rt_8, val_8)

    def _srch_8_rec(self, nd_8, val_8):
        if nd_8 is None or nd_8.val_8 == val_8:
            return nd_8
        if val_8 < nd_8.val_8:
            return self._srch_8_rec(nd_8.lhs_8, val_8)
        return self._srch_8_rec(nd_8.rhs_8, val_8)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g1/renamed_44_bst.java
class Node_8 {
    int val_8;
    Node_8 lhs_8, rhs_8;
    public Node_8(int item) {
        val_8 = item;
        lhs_8 = rhs_8 = null;
    }
}
public class Tree_8 {
    public static void main(String[] args) {
        Tree_8 bst = new Tree_8();
        bst.ins_8(50);
        bst.srch_8(50);
    }
    Node_8 rt_8;
    public void ins_8(int val_8) {
        rt_8 = in_rec_8(rt_8, val_8);
    }
    Node_8 in_rec_8(Node_8 rt_8, int val_8) {
        if (rt_8 == null) return new Node_8(val_8);
        if (val_8 < rt_8.val_8) rt_8.lhs_8 = in_rec_8(rt_8.lhs_8, val_8);
        else if (val_8 > rt_8.val_8) rt_8.rhs_8 = in_rec_8(rt_8.rhs_8, val_8);
        return rt_8;
    }
    Node_8 srch_8(int val_8) {
        return sr_rec_8(rt_8, val_8);
    }
    Node_8 sr_rec_8(Node_8 rt_8, int val_8) {
        if (rt_8 == null || rt_8.val_8 == val_8) return rt_8;
        if (rt_8.val_8 < val_8) return sr_rec_8(rt_8.rhs_8, val_8);
        return sr_rec_8(rt_8.lhs_8, val_8);
    }
}